dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -45,393 +58,492 @@ public class DidDhtConfiguration internal constructor(
+   * parsed, and used to reconstruct the DID Document.
+   *
+   * @param did The "did:dht" DID that needs to be resolved.
+   * @return A [DidResolutionResult] instance containing the DID Document and related context, including types
+   * as part of the [DidDocumentMetadata], if available.
+   * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+   */
+  override fun resolve(did: String, options: ResolveDidOptions?): DidResolutionResult {
+    validate(did)
+    val getId = DidDht.suffix(did)
+    val bep44Message = dht.pkarrGet(getId)
+    val dnsPacket = Dht.parseBep44GetResponse(bep44Message)
+    fromDnsPacket(did, dnsPacket).let { (didDocument, types) ->
+      return DidResolutionResult(didDocument = didDocument, didDocumentMetadata = DidDocumentMetadata(types = types))
     }
+  }
 
-    /**
-     * Converts a [DIDDocument] to a DNS packet according to the did:dht spec
-     * https://tbd54566975.github.io/did-dht-method/#dids-as-a-dns-packet
-     *
-     * @param didDocument The [DIDDocument] to convert.
-     * @param types A list of types to include in the packet.
-     * @return A [Message] instance containing the DNS packet.
-     */
-    public fun toDnsPacket(didDocument: DIDDocument, types: List<Int>? = null): Message {
-      val message = Message(0).apply { header.setFlag(5) } // Set authoritative answer flag
-
-      // map key ids to their verification method ids
-      val verificationMethodsById = mutableMapOf<String, String>()
-
-      // track all verification methods and services by their ids
-      val verificationMethodIds = mutableListOf<String>()
-      val serviceIds = mutableListOf<String>()
-
-      // Add Resource Records for each Verification Method
-      didDocument.verificationMethods?.forEachIndexed { i, verificationMethod ->
-        val publicKeyJwk = JWK.parse(verificationMethod.publicKeyJwk)
-        val publicKeyBytes = Crypto.publicKeyToBytes(publicKeyJwk)
-        val base64UrlEncodedKey = Base64URL(Convert(publicKeyBytes).toBase64Url(padding = false))
-        val vmId = "k$i"
-
-        verificationMethodsById[verificationMethod.id.toString()] = vmId
-
-        val keyType = when (publicKeyJwk.algorithm) {
-          JWSAlgorithm.EdDSA -> 0
-          JWSAlgorithm.ES256K -> 1
-          else -> throw IllegalArgumentException("unsupported algorithm: ${publicKeyJwk.algorithm}")
-        }
+  /**
+   * Publishes a [DIDDocument] to the DHT.
+   *
+   * @param manager The [KeyManager] instance to use for signing the message.
+   * @param didDocument The [DIDDocument] to publish.
+   * @param types A list of types to include in the packet.
+   * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+   * @throws Exception if the message is not successfully put to the DHT.
+   */
+  public fun publish(manager: KeyManager, didDocument: DIDDocument, types: List<Int>? = null) {
```

Comment:
can we use an enum for the possible `types`?

Reply:
added